lib/http-invocation.js

```diff
@@ -223,14 +229,37 @@ HttpInvocation.prototype.invoke = function(callback) {
   if (!this.req) {
     this.createRequest();
   }
-  request(this.req, function(err, res, body) {
-    if (err instanceof SyntaxError) {
-      if (res.status === 204) err = null;
+  var method = this.method;
+  var streamDesc = method.sharedMethod.stream;
+  var methodReturnsStream = !!streamDesc;
+  // var streamResult;
+
+  if (methodReturnsStream) {
+    if (streamDesc.type === 'ReadableStream' && streamDesc.json) {
+      var mdm = new MuxDemux();
+      mdm.on('connection', function(stream) {
+        callback(null, stream);
+      });
+
+      request(this.req).pipe(mdm);
+
+      // streamResult = request(this.req).pipe(JSONStream.parse());
+
+    } else {
+      callback(new Error('unsupported stream type'));
     }
-    if (err) return callback(err);
-    self.res = res;
-    self.transformResponse(res, body, callback);
-  });
+
+    return;
+  } else {
```

Comment:
It may be my personal preference, but I would omit unnecessary `else` block after `return` and thus keep the amount of whitespace changes lower.